src/Microsoft.Performance.Toolkit.Engine/EngineException.cs

```diff
@@ -628,4 +628,98 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
             base.GetObjectData(info, context);
         }
     }
+
+    /// <summary>
+    ///     Represents the error that occurs when an attempt is made to enable a
+    ///     source cooker on the engine when the data sources do not provide the
+    ///     data required for said cooker.
+    ///     <para />
+    ///     An example of this condition would be when you attempt to add a cooker
+    ///     that cooks events from an ETL file (Event Tracing for Windows trace file),
+    ///     but no ETL files are present in the set of files to process.
+    /// </summary>
+    [Serializable]
+    public class NoInputDataException
```

Comment:
>NoInputDataException [](http://example.com/codeflow?start=17&length=20)

Should this be NoDataSourceException?

Reply:
I was trying to communicate that the source cooker did not have any input. I can rename, I think that makes sense.